### DISCUSSION TITLE
Unexpectedly high CPU load on macOS

### DISCUSSION THREAD

**Discussion by anaxci** [iterate-ch/cyberduck#14894] [Q&A] [answered]
When using Mountainduck I'm constantly having very high CPU load between 150% and 750%. 
I thought it might be a lot of traffic in the background but even when there is no connection established to any server the load is really high.
Is this a common issue?

**Comment by Mr-Alexander** [comment]
i am having the same issue, CPU 600-1,000%, even after I disconnect from any open connection.

![Screenshot 2023-07-10 at 16 45 51](https://github.com/iterate-ch/cyberduck/assets/107499301/75104711-6529-40f0-8848-16a47676ea89)

**Comment by anaxci** [comment]
Also with the latest update the issue persists

**Comment by harrisdan** [comment]
The same is happening for me.   It's been a persistent problem for several app updates, but I've finally grown tired of it and am trying to find a solution.  I will mount my SFTP drives, and when I first come in to work the next day, I'll see Mountain Duck using 600% CPU until I kill it and remount.  I submitted a log file to support via e-mail but am adding my "me too" here.

**Comment by Mr-Alexander** [comment]
i thought that the issue was an intel mac but it happens on my M1 Max and M1 Ultra macs as well..

`1,734.8 %`

that's insane...

<img width="592" alt="Screenshot 2023-10-12 at 17 25 29" src="https://github.com/iterate-ch/cyberduck/assets/107499301/ccdd9dee-6045-4d0d-a441-d3c86d07388f">

**Comment by dkocher** [comment, marked as answer]
There are several reasons that can cause unexpectedly high CPU usage but in general it indicates an application is accessing files on the mounted volume very frequently. For example this could be:
* An Antivirus application scanning files on the mounted volume. Make sure to exclude the [mount location](https://docs.mountainduck.io/mountainduck/preferences/#mount-location) from your AV software if any.
* Spotlight on macOS indexing files on the volume. This should be prevented with the suffix `.noindex` on the mount volume path set by default.
* A backup application such as _Backblaze Backup_
* [High CPU usage when working with Sublime Text](https://docs.mountainduck.io/mountainduck/issues/#high-cpu-usage-when-working-with-sublime-text)

Another culprit can be the _Index Files_ option set in _Preferences → Sync_ for volumes with a large folder structure with a remote server where on cannot determine if contents of folders has changed. This applies to object storage such as S3. 

When connected to SFTP volumes, make sure there are no symbolic links that point to a parent or working directory causing folders to be indexed recursively.

When all the above does not apply please send a thread dump to [support](mailto:support@mountainduck.io):

1. Open Terminal.app
2. Enter 

```
kill -USR1 $(ps aux|grep "[M]ountain Duck"|awk '{print $2}')
```
3. Choose _Preferences → Connection → Log → Show In Finder_ from the Mountain Duck status bar menu.
4. Attach the file `mountainduck.dmp`

**Comment by svenniemann** [comment]
What was the actual solution here?
I also have this problem. Random high CPU. After closing Mountain Duck and reopen it is fine again. Until it happens the next time. Running latest snapshot version.

**Comment by dkocher** [comment]
Relates to [^1].

[^1]: #16431

**Comment by Tezzlicious** [comment]
**Latest update (01/05/2025):**

After some emails with David the high CPU consumption is expected when syncing folders with lots of files in Smart Sync mode. Online mode doesn't seem to suffer this as much, because it fetches only what is needed at that moment to the cache. It still consumes CPU cycles and I can only assume that the team already does this in the most efficient way possible.

Some snippets from the emails:

> Downloads are performed with multiple segments in parallel which may cause the system to show CPU spikes when merging the segmented file contents.
> 
> You may want to disable this feature using a hidden configuration option as documented in https://docs.mountainduck.io/tutorials/hidden_properties/#using-default-properties
> 
> `queue.download.segments=false`
> 


> I can confirm that resuming sync after restarting Mountain Duck with many items in the queue can cause quite some performance penalty as all file paths are checked for validity.


This is not only after restarting, but every sync with many items the CPU utilization goes up and progress is very slow especially with many smaller files. I think this is due the nature that a remote share will never act as fast as local shares because of several reasons.

I also experienced stalls during my observations, but I really can't bother (and don't have time) to find out what is causing this.
So for my use cases I decided remote mounting of shares is not a usable option and bought an external USB-C SSD drive. 

**Original post:**

Please find whatever is causing this. It renders Mountain Duck unusable on my Macbook M2 Max. I followed every tip that I can find, still goes haywire after a certain period of usage.

Things I tried already:

- Change mount point with .noindex added to the main mount folder name.
- Exclude mount point from malware/virus scanner
- No backup software is active on any of the Mountain Duck shares
- Index files set to Off
- Smart Synchronization or Online, does not seem to make a difference

It starts of quietly but after some usage of any of the shares the CPU usage goes through the roof.
